zerver/tests/test_queue_worker.py

```diff
@@ -538,6 +538,7 @@ def consume(self, data: Mapping[str, Any]) -> None:
             worker = UnreliableWorker()
             worker.setup()
             with patch('logging.exception') as logging_exception_mock:
+                # Better to use patch() as we can check for stack info too.
```

Comment:
We could still use `assertLogs` and use the `records` attribute which has a list of [`logging.LogRecord`](https://docs.python.org/3/library/logging.html#logrecord-attributes) to check `stack_info`. If you've checked it out already, does it have any issues that block us from using it?